### PR TITLE
Handle dashboard status check when actor is the donor

### PR DESCRIPTION
**Pull request by hawx**
## Purpose

Fix donors who added LPAs being able to see the dashboard. 

Fixes UML-679

## Approach

The response for a donor doesn't seem to include `systemStatus`, which was naively being checked to see if the actor was active. I've changed it to check whether the actor is a donor first.

## Learning

_Any tips and tricks, blog posts or tools which helped you. Plus anything notable you've discovered about the Refunds service_

## Checklist

* [ ] I have performed a self-review of my own code
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [ ] I have added tests to prove my work
* [ ] The product team have tested these changes

### CODE REVIEW

File: test-data/api-gateway.json [hunk 1]
```diff
@@ -27,8 +27,7 @@
           "addressLine3": ""
         }
       ],
-      "companyName": null,
-      "systemStatus": true
```
Comment (cooperaj) [inline test-data/api-gateway.json:31]: This data is what we get back from the sirius api. Have they changed that?

Comment (hawx) [reply test-data/api-gateway.json:31]: This is what I got when I got some earlier today

Comment (hawx) [reply test-data/api-gateway.json:31]: in my head it makes sense: the donor is always active

Comment (cooperaj) [reply test-data/api-gateway.json:31]: Guess they changed that then.

Comment (hawx) [reply test-data/api-gateway.json:31]: Yeah, I've had a look through the Sirius code and I think the swagger is wrong. It says that both [`donor`](https://github.com/ministryofjustice/opg-sirius/blob/e99cd97265a65609de53663547e6eeda59c3cb5b/back-end/docs/api/swagger/api.public.v1.yaml#L254) and [`attorneys`](https://github.com/ministryofjustice/opg-sirius/blob/e99cd97265a65609de53663547e6eeda59c3cb5b/back-end/docs/api/swagger/api.public.v1.yaml#L258) are `CaseActor`s, which should include [`systemStatus`](https://github.com/ministryofjustice/opg-sirius/blob/e99cd97265a65609de53663547e6eeda59c3cb5b/back-end/docs/api/swagger/api.public.v1.yaml#L325).

But the code eventually returns an `Lpa` where:
- the `donor` is of type [`Donor`](https://github.com/ministryofjustice/opg-sirius/blob/e99cd97265a65609de53663547e6eeda59c3cb5b/back-end/module/Opg/src/Opg/Core/Model/Entity/CaseActor/Donor.php), with no `systemStatus`
- but `attorneys` is a list of `Attorney`, extending [`AttorneyAbstract`](https://github.com/ministryofjustice/opg-sirius/blob/e99cd97265a65609de53663547e6eeda59c3cb5b/back-end/module/Opg/src/Opg/Core/Model/Entity/CaseActor/AttorneyAbstract.php#L33) which has `systemStatus`

**Pull request merged**